sawmill-core/src/test/java/io/logz/sawmill/processors/UrlDecodeProcessorTest.java

```diff
@@ -0,0 +1,86 @@
+            "      }\n" +
+            "    ]\n" +
+            "  },\n" +
+            "\"url\": \"https://github.com/logzio/sawmill/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed\",\n" +
+            "\"id\": \"1%20%3A\"\n" +
+            "}";
+
+
+    @Test
+    public void testAllFieldsAndCodec() throws InterruptedException {
+        String encoding = "ibm856";
+
+        Map<String, Object> config = createConfig("allFields","true","charSet",encoding);
+
+        UrlDecodeProcessor urlDecodeProcessor = createProcessor(UrlDecodeProcessor.class, config);
+
+        Doc processedDoc = new Doc(JsonUtils.fromJsonString(Map.class,messageExample));
+
+        assertThat(urlDecodeProcessor.process(processedDoc).isSucceeded()).isTrue();
+
+        Doc unprocessedDoc = new Doc(JsonUtils.fromJsonString(Map.class,messageExample));
+
+        assertThat((String) processedDoc.getField("url")).isEqualTo(decodeUrl(unprocessedDoc.getField("url"), encoding));
+        assertThat((String) processedDoc.getField("id")).isEqualTo(decodeUrl(unprocessedDoc.getField("id"),encoding));
+        assertThat((String) processedDoc.getField("innerObject.url")).isEqualTo(decodeUrl(unprocessedDoc.getField("innerObject.url"),encoding));
+        assertThat((String) processedDoc.getField("innerObject.anotherUrl")).isEqualTo(decodeUrl(unprocessedDoc.getField("innerObject.anotherUrl"),encoding));
+
+        List<Map<String,Object>> processedfriends = ((List)processedDoc.getField("innerObject.friends"));
+        List<Map<String,Object>> unProcessedfriends = ((List)unprocessedDoc.getField("innerObject.friends"));
+
+        for(int i = 0;i<processedfriends.size();++i){
+            assertThat(processedfriends.get(i).get("url")).isEqualTo(decodeUrl((String) unProcessedfriends.get(i).get("url"),encoding));
+            assertThat(processedfriends.get(i).get("id")).isEqualTo(decodeUrl((String)unProcessedfriends.get(i).get("id"),encoding));
+        }
+    }
+    @Test
+    public void testSingleFieldAndDefaultEncoding() throws InterruptedException {
+        String encoding = "UTF-8";
+        Map<String, Object> config = createConfig("field","innerObject.url");
+        UrlDecodeProcessor urlDecodeProcessor = createProcessor(UrlDecodeProcessor.class, config);
+        Doc processedDoc = new Doc(JsonUtils.fromJsonString(Map.class,messageExample));
+        assertThat(urlDecodeProcessor.process(processedDoc).isSucceeded()).isTrue();
+        Doc unprocessedDoc = new Doc(JsonUtils.fromJsonString(Map.class,messageExample));
+        assertThat((String) processedDoc.getField("innerObject.url")).isEqualTo(decodeUrl(unprocessedDoc.getField("innerObject.url"),encoding));
+    }
+
+    private String decodeUrl(String valueToDecode,String charSet) {
+        String decodedUrl = valueToDecode;
+        try{
```

Comment:
this try catch is redundant, you control your test so just use URLDecoder.decode instead of wrapping it in method